ext/featurefmt/rpm/cpe_namespace.go

```diff
@@ -63,17 +64,67 @@ func (l lister) getPotentialNamespace(files tarutil.FilesMap) (namespaces []data
 	return
 }
 
+func findContentManifestFile(files tarutil.FilesMap) string {
+	for filePath := range files {
+		if contentManifestRegex.MatchString(filePath) {
+			return filePath
+		}
+	}
+	return ""
+}
+
 // findDockerfile in filemap - RedHat stores Dockerfiles into /root/buildinfo/
 func findDockerfile(files tarutil.FilesMap) string {
 	for filePath := range files {
-		if !strings.HasPrefix(filePath, "root/buildinfo/Dockerfile-") {
-			continue
+		if dockerfileRegex.MatchString(filePath) {
+			return filePath
 		}
-		return filePath
 	}
 	return ""
 }
 
+func (l lister) getCPEsFromContentManifestFile(files tarutil.FilesMap) []string {
+	manifestFilePath := findContentManifestFile(files)
+	if manifestFilePath == "" {
+		return []string{}
+	}
+	log.WithField("path", manifestFilePath).Debug("Found content manifest in layer")
+	contentManifestBytes := files[manifestFilePath]
+	contentManifestData := contentmanifest.ContentManifest{}
+	err := json.Unmarshal(contentManifestBytes, &contentManifestData)
```

Comment:
consider a json.Decoder to stream the bytes already in memory and ont create a copy.

Reply:
Done